components/auth/Join.tsx

```diff
@@ -91,15 +95,27 @@ const Join = () => {
           error={formik.touched.email ? formik.errors.email : undefined}
           onChange={formik.handleChange}
         />
-        <InputWithLabel
-          type="password"
-          label={t('password')}
-          name="password"
-          placeholder={t('password')}
-          value={formik.values.password}
-          error={formik.touched.password ? formik.errors.password : undefined}
-          onChange={formik.handleChange}
-        />
+        <div className="relative flex">
+          <InputWithLabel
+            type={isPasswordVisible ? 'text' : 'password'}
+            label={t('password')}
+            name="password"
+            placeholder={t('password')}
+            value={formik.values.password}
+            error={formik.touched.password ? formik.errors.password : undefined}
+            onChange={formik.handleChange}
+          />
+          <button
+            onClick={handlepasswordvisibility}
```

Comment:
```suggestion
            onClick={hadlePasswordVisibility}
```

Reply:
Add property `type='button'` to the button, otherwise it will submit the form when click the icon.